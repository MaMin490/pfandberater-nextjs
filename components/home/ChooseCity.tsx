import React from 'react';
import styled from 'styled-components';
import { Menu, Dropdown, message } from 'antd';
import { DownOutlined } from '@ant-design/icons';

import { theme } from '../../public/global/colors';

const { colors } = theme;

const ChooseCityContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  color: ${colors.menuGrey};
  transform: translate(-50%, 6rem);
`;

export default function ChooseCity(): JSX.Element {
  const menu = (
    <Menu style={{ background: colors.lightGrey, color: colors.menuGrey }} onClick={onClick}>
      <Menu.Item style={{ color: colors.menuGrey }} key="1">
        augsburg
      </Menu.Item>
      <Menu.Item key="2">münchen</Menu.Item>
      <Menu.Item key="3">berlin</Menu.Item>
    </Menu>
  );

  function onClick({ key }) {
    message.info(`Click on item ${key}`);
  }

  return (
    <ChooseCityContainer>
      <Dropdown overlay={menu}>
        <button
          style={{ color: 'blue', height: '100%', width: '30rem' }}
          //   className="ant-dropdown-link"
          onClick={e => e.preventDefault()}>
          wähle deine stadt
          <DownOutlined />
        </button>
      </Dropdown>
    </ChooseCityContainer>
  );
}
