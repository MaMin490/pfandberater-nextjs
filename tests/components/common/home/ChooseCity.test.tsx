import React from 'react';
import { shallow } from 'enzyme';

import ChooseCity from '../../../../components/home/ChooseCity';

describe('ChooseCity', () => {
  it('should render correctly', () => {
    const component = shallow(<ChooseCity />);

    expect(component).toMatchSnapshot();
  });
});
