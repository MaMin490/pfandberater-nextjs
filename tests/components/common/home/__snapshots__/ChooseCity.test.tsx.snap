// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseCity should render correctly 1`] = `
<styled.div>
  <div>
    wähle deine
  </div>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Styled(Menu)
        onClick={[Function]}
      >
        <Styled(MenuItem)>
          augsburg
        </Styled(MenuItem)>
        <Styled(MenuItem)>
          münchen
        </Styled(MenuItem)>
        <Styled(MenuItem)>
          berlin
        </Styled(MenuItem)>
      </Styled(Menu)>
    }
    placement="bottomCenter"
  >
    <styled.button
      onClick={[Function]}
    >
      stadt
      <ForwardRef(DownOutlined)
        style={
          Object {
            "fontSize": "1.3rem",
          }
        }
      />
    </styled.button>
  </Dropdown>
</styled.div>
`;
